x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_header/rename_conversation_input.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonIcon, EuiFieldText, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React, { useCallback, useRef, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import { useConversationContext } from '../../../context/conversation/conversation_context';
+import { useConversationId } from '../../../context/conversation/use_conversation_id';
+import { useConversationTitle } from '../../../hooks/use_conversation';
+
+const labels = {
+  inputPlaceholder: i18n.translate(
+    'xpack.onechat.conversationTitle.renameConversationInput.inputPlaceholder',
+    {
+      defaultMessage: 'Enter conversation name',
+    }
+  ),
+  confirmButton: i18n.translate(
+    'xpack.onechat.conversationTitle.renameConversationInput.confirmButton',
+    {
+      defaultMessage: 'Confirm rename',
+    }
+  ),
+  cancelButton: i18n.translate(
+    'xpack.onechat.conversationTitle.renameConversationInput.cancelButton',
+    {
+      defaultMessage: 'Cancel rename',
+    }
+  ),
+};
+
+const INPUT_WIDTH = '240px';
+
+const inputWidthStyles = css`
+  width: ${INPUT_WIDTH};
+`;
+
+interface RenameConversationInputProps {
+  onCancel: () => void;
+}
+
+export const RenameConversationInput: React.FC<RenameConversationInputProps> = ({ onCancel }) => {
+  const conversationId = useConversationId();
+  const { title } = useConversationTitle();
+  const { conversationActions } = useConversationContext();
+  const [isLoading, setIsLoading] = useState(false);
+  const [newTitle, setNewTitle] = useState(title || '');
+  const hasFocusedRef = useRef(false);
+
+  // Callback ref that focuses only on first mount
+  const inputRefCallback = useCallback((el: HTMLInputElement | null) => {
+    if (el && !hasFocusedRef.current) {
+      hasFocusedRef.current = true;
+      // Use requestAnimationFrame to ensure the element is fully mounted
+      requestAnimationFrame(() => {
+        el.focus();
+        el.select();
+      });
+    }
+  }, []);
+
+  const isFormDirty = newTitle.trim() !== (title || '').trim();
+
+  const handleRename = useCallback(async () => {
+    if (!conversationId || !newTitle.trim() || !isFormDirty) {
+      return;
+    }
+    setIsLoading(true);
+    await conversationActions.renameConversation(conversationId, newTitle.trim());
+    setIsLoading(false);
+    onCancel();
+  }, [conversationId, newTitle, conversationActions, onCancel, isFormDirty]);
+
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      if (e.key === 'Enter') {
+        e.preventDefault();
+        if (isFormDirty && newTitle.trim()) {
+          handleRename();
+        }
+      } else if (e.key === 'Escape') {
```

Comment:
We probably have some EUI constant for getting these enter and escape keys I imagine, will revisit it on Monday